### DISCUSSION TITLE
How to know if body is a sensor using just BodyID

### DISCUSSION THREAD

**Discussion by benjaminleezhiyuan** [jrouwe/JoltPhysics#1356] [Q&A] [answered]
I have a character contact listener
```
void MyCharacterContactListener::OnContactAdded(const CharacterVirtual* inCharacter, const BodyID& inBodyID2, const SubShapeID& inSubShapeID2, RVec3Arg inContactPosition, Vec3Arg inContactNormal, CharacterContactSettings& ioSettings)
{
	/*	If body is sensor
		push into trigger queue
		else					*/
	sData.onCharacterCollisionPairAddedQueue.push({ &inCharacter, PhysicsSystem::BodyIDToUUID(inBodyID2.GetIndexAndSequenceNumber()) });
}
```
I have separate queues for character vs sensor and character vs non-sensors

This is what my normal contact listener looks like
```
void MyContactListener::OnContactAdded(const Body& inBody1, const Body& inBody2, const ContactManifold& inManifold, ContactSettings& ioSettings)
{
	if (inBody1.IsSensor() || inBody2.IsSensor())
	{
		sData.onTriggerPairAddedQueue.push({ PhysicsSystem::BodyIDToUUID(inBody1.GetID().GetIndexAndSequenceNumber()), PhysicsSystem::BodyIDToUUID(inBody2.GetID().GetIndexAndSequenceNumber()) });
		spdlog::info("Trigger pair added");
	}
	else
	{
		sData.onCollisionPairAddedQueue.push({ PhysicsSystem::BodyIDToUUID(inBody1.GetID().GetIndexAndSequenceNumber()), PhysicsSystem::BodyIDToUUID(inBody2.GetID().GetIndexAndSequenceNumber()) });
	}
}
```

I can check if body is sensor using the isSensor function
but for the character contact listener I only have the bodyID not the body.
how do you get the body through bodyID without locking interface.

**Comment by jrouwe** [comment, marked as answer]
If you don't want to lock then use something like:

```BodyLockRead lock(mPhysicsSystem.GetBodyLockInterfaceNoLock(), inBodyID2);```

(although for CharacterVirtual you're calling the update function yourself and if at that time other operations on the physics system can be done then you would need to use the locking interface)

**Reply by benjaminleezhiyuan** [reply]
I made a map of bodyID to bool that returns if its a sensor and it should work for my use case.
